vllm/platforms/__init__.py

```diff
@@ -1,23 +1,54 @@
-import torch
-
 from .interface import Platform, PlatformEnum, UnspecifiedPlatform
 
 current_platform: Platform
 
+# NOTE: we don't use `torch.version.cuda` / `torch.version.hip` because
+# they only indicate the build configuration, not the runtime environment.
+# For example, people can install a cuda build of pytorch but run on tpu.
+
+is_tpu = False
+try:
+    import torch_xla.core.xla_model as xm
+    xm.xla_device(devkind="TPU")
+    is_tpu = True
+except Exception:
+    pass
```

Comment:
cc @WoosukKwon 

we don't use any successful `import` as a flag. Instead, only when some device code executes successfully, then we trust that we are in the current platform.

technically, we can install `libtpu` python package for any platform.